### DISCUSSION TITLE
Adding dependency graph into Github private repo

### DISCUSSION THREAD

**Discussion by hubdwoo** [community/community#51920] [Repositories] [answered]
Hi! 

I'm kinda new to dependency graph and I'm not sure how does Github searches for dependencies.

I have a private repo that I enabled to have dependency graph. I want my dependencies to be tracked by insights. I'm hoping that in the future I could maybe put that repo public and someone could use 'pip install {git repo}' to install my dependencies to their environment.

How do I go around to do that?

I'm using Jupyter notebook and I've pushed 'requirements.txt' into my root repository hoping that it would be detected but it didn't. The file-size for the 'requirements.txt' isn't that big it's just around 200KB so it shouldn't be blocked by Github.

Any help would be appreciated!

**Comment by hubdwoo** [comment, marked as answer]
I found the source of the problem. Turns out the dependency graph only detects text that are UTF-8 encoding. The requirements.txt that I pushed was UTF-16L.

**Reply by adivardhan1000** [reply]
This is the way!!
